Lombiq.Hosting.Tenants.FeaturesGuard/Handlers/FeaturesEventHandler.cs

```diff
@@ -13,215 +12,118 @@ namespace Lombiq.Hosting.Tenants.FeaturesGuard.Handlers;
 
 public sealed class FeaturesEventHandler : IFeatureEventHandler
 {
-    private readonly IShellFeaturesManager _shellFeaturesManager;
     private readonly IOptions<ConditionallyEnabledFeaturesOptions> _conditionallyEnabledFeaturesOptions;
-    private readonly ShellSettings _shellSettings;
 
-    public FeaturesEventHandler(
-        IShellFeaturesManager shellFeaturesManager,
-        IOptions<ConditionallyEnabledFeaturesOptions> conditionallyEnabledFeaturesOptions,
-        ShellSettings shellSettings)
-    {
-        _shellFeaturesManager = shellFeaturesManager;
+    public FeaturesEventHandler(IOptions<ConditionallyEnabledFeaturesOptions> conditionallyEnabledFeaturesOptions) =>
```

Comment:
I'd resolve this from inside the deferred task too, then.

Reply:
This is actually used in the helper method which doesn't have a scope instance.
![image](https://github.com/Lombiq/Hosting-Tenants/assets/92299130/50a727d8-f0a5-4869-ab46-03216ad06503)

If you mean to resolve this in the deferredTask and then pass it as a parameter to the `GetFeaturesToBeEnabledAndDisabled` then it's okay.

Reply:
But resolving it in the helper method would require another service to get service which kills the point imo.

Reply:
You can pass the resolved options to the method, yes.